specs/commandline/blob.md

```diff
@@ -0,0 +1,381 @@
+
+# sign a blob
+notation blob sign /tmp/my-blob.bin
+```
+
+An example for a successful signing:
+
+```console
+$ notation blob sign /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file written to ./my-blob.bin.sig.jws
+```
+
+### Sign a blob by generating the detached signature in a particular directory
+```shell
+$ notation blob sign --signature-directory /tmp/xyz/sigs /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file written to /tmp/xyz/sigs/my-blob.bin.sig.jws
+```
+
+### Sign a blob and skip user confirmations when overwriting existing signature
+```shell
+$ notation blob sign --force /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file overwritten to ./my-blob.bin.sig.jws
+```
+
+### Sign a blob with on-demand remote key
+
+```shell
+notation blob sign --plugin <plugin_name> --id <remote_key_id> /tmp/my-blob.bin
+```
+
+### Sign a blob using COSE signature format
+
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+# Use option "--signature-format" to set the signature format to COSE.
+notation blob sign --signature-format cose /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file written to ./my-blob.bin.sig.cose
+```
+
+### Sign a blob using the default signing key
+
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+notation blob sign /tmp/my-blob.bin
+```
+
+### Sign a blob with user metadata
+
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+# sign a blob and add user-metadata io.wabbit-networks.buildId=123 to the payload
+notation blob sign --user-metadata io.wabbit-networks.buildId=123 /tmp/my-blob.bin
+
+# sign a blob and add user-metadata io.wabbit-networks.buildId=123 and io.wabbit-networks.buildTime=1672944615 to the payload
+notation blob sign --user-metadata io.wabbit-networks.buildId=123 --user-metadata io.wabbit-networks.buildTime=1672944615 /tmp/my-blob.bin
+```
+
+### Sign a blob and specify the media type for the blob
+
+```shell
+notation blob sign --media-type <media type> /tmp/my-blob.bin
+```
+
+### Sign a blob and specify the signature expiry duration, for example 24 hours
+
+```shell
+notation blob sign --expiry 24h /tmp/my-blob.bin
+```
+
+### Sign a blob using a specified signing key
+
+```shell
+# List signing keys to get the key name
+notation key list
+
+# Sign a container image using the specified key name
+notation blob sign --key <key_name> /tmp/my-blob.bin
+```
+
+## Inspect detached blob signatures
+
+### Display details of the given detached blob signature and its associated certificate properties
+
+
+```text
+notation blob inspect [flags] /tmp/my-blob.bin.sig.jws
+```
+
+### Inspect the given detached blob signature
+
+```shell
+# Prerequisites: Signatures is produced by notation blob sign command
+notation blob inspect /tmp/my-blob.bin.sig.jws
```

Comment:
Just curious, how does Notation tell apart signatures generated by `blob sign` and `sign` again? In other words, is there a fundamental difference that's ALWAYS making `detached signature` and `OCI signature` mutually exclusive? 

Because from the spec [PR](https://github.com/notaryproject/specifications/pull/283/files#diff-a4742f16cbae7ff1d39bafc711c835c1133dea5722c1922e522022ebdefac9f5), it seems that given a signature file, it is possible that it's both a valid `detached signature` and a valid `OCI signature`. If I pull an `OCI signature` from a registry and save it as a file, then I use `notation blob inspect` to inspect it, would Notation return an error?

Reply:
> If I pull an OCI signature from a registry and save it as a file, then I use notation blob inspect to inspect it, would Notation return an error?

No. And that is the right behavior and both OCI and Blob signatures follow the same scheme. 

>  I think for inspect, we could just add a new flag to cover all the scenarios: notation inspect --blob /tmp/my-blob.bin.sig.jws

The signature scheme is not different but aspects like signature format type, OCI signature manifest vs no manifest for blobs, etc. are couple of differences. It was decided in the Notary meeting to introduce `blob` sub-command to tackle these differences.